dashboard/app/controllers/scripts_controller.rb

```diff
@@ -143,6 +143,26 @@ def instructions
     render 'levels/instructions', locals: {stages: script.lessons}
   end
 
+  def vocab
+    unit = Script.get_from_cache(params[:script_id])
+    return render :forbidden unless unit.is_migrated
```

Comment:
it looks like this is adding a few too many new ways to do things:
1. Script.get_from_cache is already done by set_script
2. permission-checking (pilot, is_migrated) now split between here, set_script, and ability.rb

For this PR, I would suggest addressing (2) as follows:
```suggestion
    return render :forbidden unless can? :read, unit
```